ethereum/contracts/DOTApp.sol

```diff
@@ -72,26 +73,29 @@ contract DOTApp is FeeController, AccessControl {
                 _channelId == ChannelId.Incentivized,
             "Invalid channel ID"
         );
-        token.burn(msg.sender, _amount, abi.encodePacked(_recipient));
+        token.burn(msg.sender, _amount);
 
         OutboundChannel channel =
             OutboundChannel(channels[_channelId].outbound);
 
         bytes memory call = encodeCall(msg.sender, _recipient, _amount);
         channel.submit(msg.sender, call);
+
+        emit Burned(msg.sender, _recipient, _amount);
```

Comment:
We should move the `emit Burned(...)` to right below the call to `token.burn(...)`. This will make the order in which the events  are logged the same as the order when using ERC777.

Reply:
I don't think it matters that much, and the convention in Substrate and Solidity seems to be that events are emitted last in each dispatchable.